WordPress/pattern-directory
# Tools: Update packages & formatting config

This PR contains a few changes related to updating packages. It would probably be easier to review by commit, or to filter out the changes in `public_html/wp-content/themes/pattern-directory/src` (formatting). The main purpose of this change is to fix the conflicts between `lint:js` and `format:js`. See https://github.com/WordPress/gutenberg/pull/31561 for more details.

- https://github.com/WordPress/pattern-directory/commit/5d5937c11c538104926886190a6d33f35b50623a Update all outdated JS packages. This includes an update to `dart-sass`, which introduces [a deprecation warning for `/` as division](https://sass-lang.com/documentation/breaking-changes/slash-div). The code in question is from the wporg parent theme, so the warning can be ignored.
- https://github.com/WordPress/pattern-directory/commit/3ee682158d202e783cfdb8ea117d74ea53ebab8b Switch to the maintained version of grunt-postcss, `@lodder/grunt-postcss`, so that we can support PostCSS 8.
- https://github.com/WordPress/pattern-directory/commit/c3f17a37a9b729e0b199b2a61574313671ed5efd Use the `recommended` lint rule which uses the prettier formatting, so no more conflicts between lint & format (see https://github.com/WordPress/pattern-directory/pull/95). Also, `wp-scripts format-js` was changed to `wp-scripts format`.
- https://github.com/WordPress/pattern-directory/commit/f1ec51ec8cc2a3a1cea488c0314bc3d7facb7c1b Add the prettier config into the local eslint config, so that local eslint (like in code editors) can understand.

### Screenshots

No visual changes.

### How to test the changes in this Pull Request:

The actions on the PR should pass.

1. Run `yarn` to update locally
2. Run `yarn workspaces run build` and make sure the build works
3. Try `yarn workspaces run lint:js` and `yarn workspaces run format:js`, they should run without issue and make no new changes
